src/praisonai-bot/praisonai_bot/bots/_outbox.py

```
@@ -431,6 +446,18 @@ async def drain(
             try:
                 # Attempt delivery
                 payload = json.loads(entry.payload)
+                # Honest at-least-once: a recovered entry re-dispatched without
+                # positive reconciliation may be a duplicate, so let the caller
+                # visibly label it. Applied only on this unreconciled recovered
+                # branch — never to fresh pending/failed or reconciled entries.
+                if entry.status == "recovered" and recovery_annotator is not None:
+                    try:
+                        payload = recovery_annotator(entry, payload)
+                    except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)